drivers/sensor/fxos8700/fxos8700_trigger.c

```diff
@@ -136,6 +136,54 @@ static void fxos8700_work_handler(struct k_work *work)
 }
 #endif
 
+int fxos8700_set_odr(struct device *dev, s32_t dr)
+{
+	const struct fxos8700_config *config = dev->config->config_info;
+	struct fxos8700_data *data = dev->driver_data;
+
+	if (dr >= 800) {
+		dr = FXOS8700_CTRLREG1_DR_RATE_800;
```

Comment:
This logic isn't correct when the sensor is in hybrid mode (accelerometer + magnetometer). The sample rates are halved for each encoding.